We.Sell.Bread.API.Unit.Tests/Tests/Services/ProductServiceTests.cs

```diff
@@ -19,7 +19,46 @@ public void GivenDataExistWhenRetrievingProductReturnTypeMustBeListOfProductDeta
 
             product.Should().NotBeNull();
             product.Should().BeOfType(typeof(List<ProductDetailsDto>));
-           
+
+        }
+
+        [Fact]
+        public void GivenIdShouldExistAndReturnTypeObjectProductDetailsDto()
+        {
+            ProductDetailsDto? product = _productService.GetProduct(new Guid("411fbad3-925e-4044-9269-f962641f9277"));
+            string realName = "Pumpernickel Bread";
+            decimal realPrice = (decimal)20.00;
+            string realDescription = "A dense, dark bread made from coarsely ground whole rye grains";
+            int realStock = 7;
+
+            product.Should().NotBeNull();
+            product.Should().BeOfType(typeof(ProductDetailsDto));
+            Assert.Equal(realName, product.ProductName);
+            Assert.Equal(realPrice, product.Price);
+            Assert.Equal(realDescription, product.Description);
+            Assert.Equal(realStock, product.StockQuantity);
+
+        }
+
+        [Fact]
+        public void AddingANewProductShouldReturnAProductAndShouldContainTheGivenDetails()
+        {
+            string productName = "Ciabatta";
+            decimal price = (decimal)35.00;
```

Comment:
In c# if you are declaring and assigning a variable inside a method, you do not need to specify the explicit type as the type can be derived from what you have assigned to a variable. To make your code more readable, rather use the generic  "var" type for all your local variables. e.g var productName = "Ciabatta";